hnix-store-remote/src/System/Nix/Store/Remote/Types.hs

```diff
@@ -1,37 +1,53 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE ScopedTypeVariables #-}
 module System.Nix.Store.Remote.Types (
     MonadStore
+  , StoreConfig(..)
   , Logger(..)
   , Field(..)
+  , getStoreDir
   , getLog
   , flushLog
   , gotError
-  , getError) where
+  , getError
+  , setData
+  , clearData
+  ) where
 
 
-import qualified Data.ByteString.Lazy      as LBS
+import           Data.ByteString           (ByteString)
+import qualified Data.ByteString.Lazy      as BSL
```

Comment:
`MonadStore` already returns a `[Logger]`. Maybe use strict bytestrings, to avoid lists of lists of chunks?
Reading from socket will force the entire lazy bytestring anyways.

Reply:
Makes sense, binary evokes lazy bytestrings. Switched gets to strict bytestrings, also moved `System.Nix.Utils` to `System.Nix.Store.Remote.Binary`. Some stuff still needs moving there.